src/System.Data.Common/src/System/Data/Common/DbCommand.cs

```diff
@@ -200,7 +200,7 @@ public virtual Task<int> ExecuteNonQueryAsync(CancellationToken cancellationToke
 
                 try
                 {
-                    return Task.FromResult<int>(ExecuteNonQuery());
+                    return Task.Run<int>(() => ExecuteNonQuery());
```

Comment:
@saurabh500 should make a call on whether this is a desired change.  If it is, this should be tweaked, in several ways. First, since the call to ExecuteNonQuery would now be wrapped in the Task.Run, the catch block can be removed.  Second, the cancellation registration is meant to be disposed when the operation completes, but with your change it's now being disposed of as soon as the task has been queued, which would need to be fixed.  Third, from a performance perspective, this ends up with extra allocations; rather than capturing the this point via a closure, Task.Factory.StartNew should be used to pass in the ```this``` reference, and any other state that might be needed (e.g. the cancellation registration, assuming it should only be disposed of once the operation actually completes).  In other words, all of this would end up looking something like:
```C#
Task.Factory.StartNew(s =>
{
    var t = (Tuple<DbCommand, CancellationTokenRegistration)s;
    try { t.Item1.ExecuteNonQuery(); }
    finally { t.Item2.Dispose(); }
}, Tuple.Create(this, registration), CancellationToken.None, TaskCreationOptions.DenyChildAttach, TaskScheduler.Default);
```
Same feedback would apply to the other changes as well.